zuul-core/src/main/java/com/netflix/zuul/message/http/HttpQueryParams.java

```diff
@@ -176,30 +187,34 @@ public void clear()
 
     public String toEncodedString()
     {
-        StringBuilder sb = new StringBuilder();
-        try {
-            for (Map.Entry<String, String> entry : entries()) {
-                sb.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
-                if (StringUtils.isNotEmpty(entry.getValue())) {
-                    sb.append('=');
-                    sb.append(URLEncoder.encode(entry.getValue(), "UTF-8"));
+        if (propertyFactory.getBooleanProperty(ZuulConstants.ZUUL_KEEP_ORIGINAL_QUERY_STRING_ENCODING,false).getValue()) {
+            return toString();
+        }else {
+            StringBuilder sb = new StringBuilder();
+            try {
+                for (Map.Entry<String, String> entry : entries()) {
+                    sb.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
+                    if (StringUtils.isNotEmpty(entry.getValue())) {
+                        sb.append('=');
+                        sb.append(URLEncoder.encode(entry.getValue(), "UTF-8"));
+                    }
+                    else if (isTrailingEquals(entry.getKey())) {
+                        sb.append('=');
+                    }
+                    sb.append('&');
                 }
-                else if (isTrailingEquals(entry.getKey())) {
-                    sb.append('=');
+
+                // Remove trailing '&'.
+                if (sb.length() > 0 && '&' == sb.charAt(sb.length() - 1)) {
+                    sb.deleteCharAt(sb.length() - 1);
                 }
-                sb.append('&');
             }
-
-            // Remove trailing '&'.
-            if (sb.length() > 0 && '&' == sb.charAt(sb.length() - 1)) {
-                sb.deleteCharAt(sb.length() - 1);
+            catch (UnsupportedEncodingException e) {
+                // Won't happen.
+                e.printStackTrace();
```

Comment:
Again, I know it's not your fault, but since you are touching this code:   This should be throwing an AssertionError(e), rather than printing to stderr.